src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -784,11 +784,17 @@ private int FindFinalStatePosition(ReadOnlySpan<char> input, int i, int timeoutO
                         }
                     }
 
-                    // Now run the DFA or NFA traversal from the current point using the current state.
+                    // Now run the DFA or NFA traversal from the current point using the current state. If timeouts are being checked,
+                    // we need to pop out of the inner loop every now and then to do the timeout check in this outer loop.
+                    const int CharsPerTimeoutCheck = 10_000;
```

Comment:
Do you think it is worth making this configurable? e.g., via an Env variable?

Reply:
We don't do that for the existing constant: https://github.com/dotnet/runtime/blob/7db092a6ed68a5fbf47962c45bb0c1bb965d02c6/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs#L86